web_client/src/textIntervalReader.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ *  A higher order component that reads text from `readTextInterval`: an async function that returns
+ *  `{ text, readingEvery }`, where `text` is a text string read and `readingEvery` is the number
+ *  of seconds after which new `text` should be read. Every `readingEvery` seconds, this component
+ *  will read text from `readTextInterval` and pass the read text to `WrappedComponent`s `text` prop.
+ */
+const withTextIntervalReader = (WrappedComponent, readTextInterval) => (
+    class extends Component {
+        state = {
+            text: ''
+        }
+        onTextIntervalRead({ text }) {
+            this.setState({text});
+        }
+        onFirstTextIntervalRead({ text, readingEvery }) {
+            this.intervalId = setInterval(() => {
+                readTextInterval().then(this.onTextIntervalRead.bind(this));
```

Comment:
instead of `bind` you can define (line 15) the function as an arrow function:

```js
onTextIntervalTead = (text) => {
  this.setState({ text });
}